File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        s.erase(remove(s.begin(),s.end(),' '),s.end());
        int k=0;
        for(int i=0;i<s.size();i++){
            char ch=s[i];
            if (!((ch>='a' && ch<='z') || (ch>='A' && ch<='Z') || (ch>='0' && ch<='9'))){
                s.erase(i,1);
                i--;
            } else{
                continue;
            }
        }
        int n=s.size()-1;
        while (k<=n){
            if (tolower(s[k])!=tolower(s[n])){
                return false;
            } else{
                k++;
                n--;
            }
        }
        return true;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
       if(s.length()==1){
        return true;
       }
        String ans=s.trim();
        ans=ans.toLowerCase();
        
        String ans1=ans.replaceAll("[^a-z0-9]","");
        int i=0,j=ans1.length()-1;
        while(i<j){
            if (ans1.charAt(i)!=ans1.charAt(j)){
                return false;
            } else{
                i++;
                j--;
            }
        }
        return true;
    }
}